src/libambit/device_driver_ambit3.c

```diff
@@ -579,7 +579,7 @@ static int parse_log_header(libambit_sbem0102_data_t *reply_data_object, ambit3_
         data += 19;
         break;
       case 0x8a: /* gen2+gen3 fw */
-        data += 4;
+        data += 1; // Skip size ??
```

Comment:
I know this part of code is is kind of "broken", but I think this change might break the log parsing?
Line 575 - 582 is a hack making the code backward compatible with older versions of firmware.

Have you tested it to work to get activity logs from your ambit3 with newest firmware?